### PR TITLE
mrdox cmake script for self-test.

### PR DESCRIPTION
**Pull request by klemens-morgenstern**
redux of #25, without the filename chance. I'd still pefer to have this in mrdox, but the self-test is useful nontheless.

### CODE REVIEW

**Comment by vinniefalco** [comment]
Yeah this looks interesting, but I want to understand what it does.

**Comment by vinniefalco** [comment]
I understand now. Well I want to reorganize the contents of the stuff in the tests/ directory along these lines, which I think will be helpful for this PR:
https://github.com/vinniefalco/mrdox/commit/7abcc1cf631def42479d90d9a4e1fefd6a5fc109

**Comment by klemens-morgenstern** [comment]
@vinniefalco update.

File: mrdox.cmake [hunk 1]
```diff
@@ -0,0 +1,64 @@
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# Copyright (c) 2023 Vinnie Falco (vinnie.falco@gmail.com)
```
Comment (vinniefalco) [inline mrdox.cmake:6]: your name goes here

**Pull request closed without merging**